react-component-template/index.jsx

```diff
@@ -12,19 +14,8 @@ export default class {{PascalName}} extends Component {
     name: '{{titleName}}'
   }
 
-  constructor (props) {
-    super(props)
-    // initialize state values
-    this.state = {}
-  }
-
-  // use the pure-render mixin without the mixin. This allows us to use es6
-  // classes and avoid "magic" code. NOTE: if this component is used directly
-  // by react-router, you should delete it, otherwise, the <Link> component will
-  // not cause a re-render
-  shouldComponentUpdate (...args) {
-    return shouldComponentUpdate.apply(this, args)
-  }
+  // initialize state values
+  state = {}
```

Comment:
Is there any purpose to setting state by default in an intializer? 
- 'dumb' components with no internal state (ie - the just display `props`) don't need `state`
- components that derive initial state from the props will need to set it in the constructor
- components that have a internally modified state, that don't inherit the initial state from the`props` benefit from having state set by an initializer.

It seems to me that the last case is probably going to be in the minority. IMHO, we might want to leave the constructor and state initializers to be defined after component generation.